stable/mongodb/templates/svc-primary-rs.yaml

```
@@ -14,6 +14,11 @@ metadata:
 {{- end }}
 spec:
   type: {{ .Values.service.type }}
+   {{- if eq .Values.service.type "ClusterIP" }}
```

Not tested, but... Can you try using it in a single line? Something like
```yaml
{{ if (.Values.service.clusterIP) and eq .Values.service.type "ClusterIP" }}
clusterIP: {{ .Values.service.clusterIP }}
{{- end }}

```

---

make sense will fix in next rev

---

Hi @carrodher unfortunately what you suggested is not working in all scenario. I think probably thats why I went with two step process earlier as well. Do you have any better suggestion to make it work?

I your code above if first condition is satisfying and second condition is not, still its allowing to go inside the control. Not sure why, but doesn't seems to work.

I tired few more pipelines to combine two condition, but is failing in one or other way.

Is it ok to keep it in 2 step?

---

Hi @carrodher can you please confirm and approve the pull request.

---

Hi, I am almost sure we can do it in a single if. Above solution was not testet and was to give an idea (pseudo-code), the real solution should be something like
```suggestion
{{- if and (.Values.service.clusterIP) (eq .Values.service.type "ClusterIP") }}
```
or
```suggestion
{{- if (and (.Values.service.clusterIP) (eq .Values.service.type "ClusterIP")) }}
```

Can you try it?